sky/skylet/providers/gcp/config.py

```diff
@@ -994,19 +992,26 @@ def _list_vpcnets(config, compute, filter=None):
 
 
 def _list_subnets(
-    config, compute, filter=None
+    config, compute, network=None
 ) -> List["google.cloud.compute_v1.types.compute.Subnetwork"]:
     response = (
         compute.subnetworks()
         .list(
             project=config["provider"]["project_id"],
             region=config["provider"]["region"],
-            filter=filter,
         )
         .execute()
     )
 
-    return response["items"] if "items" in response else []
+    items = response["items"] if "items" in response else []
+    if network is None:
+        return items
+    matched_items = []
+    for item in items:
+        # 'https://www.googleapis.com/compute/v1/projects/<project_id>/global/networks/<network_name>'
+        if network == item["network"].split("/")[-1]:
```

Comment:
Should we use `_network_interface_to_vpc_name`?

Reply:
Good catch, done!